docs/rbac/roles.rst

```diff
@@ -0,0 +1,275 @@
+
+* ``pulp_container/namespaces``
+* ``distributions/container/container``
+* ``repositories/container/container-push``
+* ``remotes/container/container``
+* ``repositories/container/container``
+* ``repositories/container/container-push/versions``
+* ``repositories/container/container/versions``
+* ``content/container/blobs``
+* ``content/container/manifests``
+* ``content/container/tags``
+
+
+Default Roles
+-------------
+
+For each endpoint, there is defined a different set of roles. The roles can be assigned at the model
+or object level for every user or group. In the following sections, there are introduced *Creator*,
+*Owner*, *Consumer*, and *Collaborator* roles. The *Consumer* and *Collaborator* roles are defined
+only for namespaces and distributions (i.e., container repositories served by the Pulp Registry).
+
+Creator Role
+~~~~~~~~~~~~
+
+The *Creator* role contains the ``add`` permission for objects present on a particular endpoint.
+For the endpoint which hosts distributions, only users with the ``container.add_containerdistribution``
+permission can create objects:
+
+.. code-block:: bash
+
+    pulp role show --name "container.containerdistribution_creator"
+
+::
+
+    {
+      "pulp_href": "/pulp/api/v3/roles/1a8555c8-3bfc-4688-81e3-5bf6fa38b9d7/",
+      "pulp_created": "2022-05-26T12:02:28.872667Z",
+      "name": "container.containerdistribution_creator",
+      "description": null,
+      "permissions": [
+        "container.add_containerdistribution"
+      ],
+      "locked": true
+    }
+
+To perform operations on an endpoint (aka ViewSet actions), a user may need to have additional
+permissions. One of the following *conditions* need to be satisfied to create a new distribution:
+
+.. code-block:: bash
+
+    pulp access-policy show --viewset-name "distributions/container/container" | jq -r '.statements[] | select(.action[] | contains("create"))'
+
+::
+
+    {
+      "action": [
+        "create"
+      ],
+      "effect": "allow",
+      "condition": "has_namespace_model_perms",
+      "principal": "authenticated"
+    }
+    {
+      "action": [
+        "create"
+      ],
+      "effect": "allow",
+      "condition": "has_namespace_perms:container.add_containerdistribution",
+      "principal": "authenticated"
+    }
+    {
+      "action": [
+        "create"
+      ],
+      "effect": "allow",
+      "condition": "namespace_is_username",
+      "principal": "authenticated"
+    }
+
+.. note::
+
+    A user, who has the *Creator* role for namespaces assigned, does not need to have any additional
+    roles to create distributions within the given namespaces. Similarly, the user is allowed to create
+    distributions within the owning username namespace (e.g., user ``alice`` can create container
+    repositories like ``alice/repo1``).
+
+
+Owner Role
+~~~~~~~~~~
+
+The *Owner* role contains all of the permissions available for ViewSet's objects but the
```

Comment:
```suggestion
The *Owner* role contains all of the permissions available for a ViewSet's objects but the
```

Reply:
I shall not use an indefinite article for plural forms.

Reply:
I am referring to `ViewSet's objects`. And, `objects` are in a plural form.

Reply:
Hm. I believe that "a" does refer to "ViewSet" . If I use this alternate construction "...available to **the objects** of **a ViewSet**", does that make the difference clearer?

And as always, my apologies for English...